galsim/chromatic.py

```diff
@@ -130,6 +135,210 @@ def __init__(self, obj):
+        required precision has been reached.  Also note that after calling `setupInterpolation`, it
+        is possible to revert to the exact calculation by calling `noInterpolation`.
+
+        The input parameter `waves` determines the input grid on which images are precomputed.  It
+        is difficult to give completely general guidance as to how many wavelengths to choose or how
+        they should be spaced; some experimentation compared with the exact calculation is warranted
+        for each particular application.  The best choice of settings might depend on how strongly
+        the parameters of the object depend on wavelength.
+
+        @param waves            The list, tuple, or NumPy array of wavelengths to be used when
+                                building up the grid of images for interpolation.  The wavelengths
+                                should be given in nanometers, and they should span the full range
+                                of wavelengths covered by any bandpass to be used for drawing Images
+                                (i.e., this class will not extrapolate beyond the given range of
+                                wavelengths).  They can be spaced any way the user likes, not
+                                necessarily linearly, though interpolation will be linear in
+                                wavelength.
+        @param oversample_fac   Factor by which to oversample the stored profiles compared to the
+                                default, which is to sample them at the Nyquist frequency for
+                                whichever wavelength has the highest Nyquist frequency.
+                                `oversample_fac`>1 results in higher accuracy but costlier
+                                pre-computations (more memory and time). [default: 1]
+        """
+        # Set up the interpolation (which can be a costly step, depending on the length of
+        # `waves`).
+        self.waves = np.sort(np.array(waves))
+
+        # Make the objects between which we are going to interpolate.  Note that these do not have
+        # to be saved for later, unlike the images.
+        objs = [ self.evaluateAtWavelength(wave) for wave in waves ]
+
+        # The evaluation step (above) led to the incorporation of any shears and other
+        # transformations into the stored models between which we interpolation, so reset the
+        # internal attributes that store information about transformations, if indeed there were any
+        # transformations.
+        transform_done = False
+        for wave in waves:
+            if hasattr(self, '_A') and not self._nullTransformation(wave):
+                transform_done = True
+                break
+        if transform_done and hasattr(self, '_A'):
+            self._A = lambda w: np.matrix(np.identity(3), dtype=float)
+            self._fluxFactor = lambda w: 1.0
+            self._reset = True
+
```

Comment:
Hey Rachel,
I pushed a commit that passes your unit tests (at least on my laptop).  I'm not 100% certain what was causing the problem, but I think it had something to do with `ChromaticConvolution.drawImage` creating copies of the things it was convolving and also creating entirely new objects as part if its strategy to fast track any separable components of the convolution.  I'm guessing these temporary objects did have their interpolation destroyed, but the originals did not.

So instead I tried destroying the interpolation when the transformation is first applied instead of waiting for drawImage to get called.  Seems to work!

-Josh